pkg/scale/encode.go

```diff
@@ -374,9 +516,12 @@ func (es *encodeState) encodeUint(i uint) (err error) {
 // encodeUint128 encodes a Uint128
 func (es *encodeState) encodeUint128(i *Uint128) (err error) {
 	if i == nil {
-		err = fmt.Errorf("uint128 is nil: %v", i)
+		err = fmt.Errorf("%w: nil", ErrEncodeUint128)
```

Comment:
`ErrEncodeUint128` should be changed to `ErrUint128IsNil` (or similar) and just wrapped as `rr = fmt.Errorf("%w", ErrUint128IsNil)` (since that `: nil` suffix is rather confusing.